libs/Spawn/Spawn.luau

```diff
@@ -1,23 +1,25 @@
-local FreeThread: thread? = nil
+local FreeThreads: { thread } = {}
 
-local function FunctionPasser(Callback, ...)
-	local AquiredThread = FreeThread
-	FreeThread = nil
+local function RunCallback(Callback, Thread, ...)
 	Callback(...)
-	FreeThread = AquiredThread
+	table.insert(FreeThreads, Thread)
 end
 
 local function Yielder()
 	while true do
-		FunctionPasser(coroutine.yield())
+		RunCallback(coroutine.yield())
 	end
 end
 
 return function<T...>(Callback: (T...) -> (), ...: T...)
-	if not FreeThread then
-		FreeThread = coroutine.create(Yielder)
-		coroutine.resume(FreeThread :: any)
+	local Thread
+	if #FreeThreads > 0 then
+		Thread = FreeThreads[1]
+		FreeThreads[1], FreeThreads[#FreeThreads] = FreeThreads[#FreeThreads], nil
```

Comment:
why take the first thread and replace it with the last one, instead of just taking the last one?

Reply:
It's slightly slower to do that

![image](https://github.com/red-blox/Util/assets/60469830/64dcebbf-98c7-4d5a-9925-0d36eec4b20a)

To confirm this is what you are talking about, this is what I changed it to
```lua
if #FreeThreads > 0 then
	thread = FreeThreads[#FreeThreads]
	FreeThreads[#FreeThreads] = nil
	-- FreeThreads[1], FreeThreads[#FreeThreads] = FreeThreads[#FreeThreads], nil
else
```